pkg/sfu/downtrack.go

```diff
@@ -130,8 +130,11 @@ type DownTrackStreamAllocatorListener interface {
 	// video layer bitrate availability changed
 	OnBitrateAvailabilityChanged(dt *DownTrack)
 
-	// max published video layer changed
-	OnMaxPublishedLayerChanged(dt *DownTrack)
+	// max published spatial layer changed
+	OnMaxPublishedSpatialChanged(dt *DownTrack)
+
+	// max published temporal layer changed
+	OnMaxPublishedTemporalChanged(dt *DownTrack)
```

Comment:
Because of setting default temporal layer to max, the current may never exactly match the target (unless subscriber has chosen specific temporal layer). Doing this to limit the target to max seen and triggering a re-allocation when max seen changes. This allows current to lock to target cleanly. Previous code did not affect things functionally, but there is code which does things like `if currentLayer != targetLayer` in per packet path. Why run them unnecessarily?